flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/FlinkContext.scala

```diff
@@ -28,6 +28,11 @@ import org.apache.calcite.plan.Context
   */
 trait FlinkContext extends Context {
 
+  /**
+   * Returns whether the planner runs in batch mode.
+   */
+  def isBatchMode: Boolean
```

Comment:
Because batch is a special case of streaming and actually it should be `isBatchMode` because streaming mode is the default and fallback. When introducing such a global flag I would rather go with correctness than consistency.